presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -88,7 +88,7 @@
     public void close()
             throws SQLException
     {
-        super.execute(format("DEALLOCATE PREPARE %s", statementName));
+        optionalConnection().ifPresent(x -> x.removePreparedStatement(statementName));
```

Comment:
I think we should still call `DEALLOCATE PREPARE ..` atleast once like before. Executing this statement will take care for removing it from `preparedStatements` in `PrestoConnection` as well. 

If you dont call `DEALLOCATE PREPARE` eventually you will hit a limit 
https://dev.mysql.com/doc/refman/5.7/en/deallocate-prepare.html

So I suggest we add a boolean `isClosed` in PrestoPreparedStatement and check that before going further.
```
    private boolean isClosed = false;
    
    public void close()
            throws SQLException
    {
        if (isClosed) {
            return;
        }
        super.execute(format("DEALLOCATE PREPARE %s", statementName));
        super.close();
        isClosed = true;
     }
```